AK/Error.h

```diff
@@ -0,0 +1,53 @@
+#pragma once
+
+#include <AK/Platform.h>
+
+namespace AK {
+
+template <typename T, auto NoErrorValue>
+class CONSUMABLE(unknown) Error {
+public:
+    RETURN_TYPESTATE(unknown)
+    Error()
+        : t(NoErrorValue)
+    {}
+
+    RETURN_TYPESTATE(unknown)
+    Error(T t)
+        : t(t)
+    {}
+
+    RETURN_TYPESTATE(unknown)
```

Comment:
Unknown is not right for the copy/move ctor. It should copy the "real" state from other, and then allow other to destruct "happily". But how?